addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/discovery/SensorDiscoveryService.java

```diff
@@ -0,0 +1,139 @@
+
+    JeeLinkHandler bridge;
+    MyReadingListener rl = new MyReadingListener();
+    AtomicBoolean capture = new AtomicBoolean();
+
+    /**
+     * Creates the discovery service for the given handler and converter.
+     */
+    public SensorDiscoveryService(JeeLinkHandler jeeLinkHandler) {
+        super(SUPPORTED_SENSOR_THING_TYPES_UIDS, DISCOVER_TIMEOUT_SECONDS, true);
+
+        bridge = jeeLinkHandler;
+
+    }
+
+    @Override
+    protected synchronized void startScan() {
+        if (!capture.get()) {
+            for (JeeLinkReadingConverter<?> c : SensorDefinition.getConverters(bridge)) {
+                c.addReadingListener(JeeLinkReadingListener.ALL, rl);
+            }
+
+            capture.set(true);
+        }
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        startScan();
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        if (capture.get()) {
+            capture.set(false);
+
+            for (JeeLinkReadingConverter<?> c : SensorDefinition.getConverters(bridge)) {
+                c.removeReadingListener(rl);
+            }
+        }
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        stopScan();
+    }
+
+    class MyReadingListener extends JeeLinkReadingListener<Reading<?>> {
+        @Override
+        public void handleReading(Reading<?> reading) {
+            if (capture.get()) {
+                final String id = reading.getSensorId();
+
+                List<Thing> existingThings = bridge.getThing().getThings();
+                boolean sensorThingExists = false;
+
+                for (Thing t : existingThings) {
+                    sensorThingExists = sensorThingExists
+                            || id.equals(t.getConfiguration().getProperties().get(PROPERTY_SENSOR_ID));
+                }
+
+                if (!sensorThingExists) {
+                    SensorDefinition def = SensorDefinition.getSensorDefinition(reading);
+                    logger.debug("read unknown sensor of type {} with id {}", def.getThingTypeUID(), id);
+
+                    boolean idExists = idExistsAtBridge(id);
+                    String newId = id;
+
+                    if (idExists) {
+                        logger.debug("bridge already has a connected sensor with thing id {}", id);
+
+                        int idx = 1;
+                        while (idExists) {
+                            newId = id + "-" + idx++;
+                            idExists = idExistsAtBridge(newId);
+                        }
+
+                        logger.debug("Using thing id {} instead", newId);
+                    }
+
+                    ThingUID sensorThing = new ThingUID(def.getThingTypeUID(), newId);
+
+                    DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(sensorThing)
+                            .withLabel(def.getName() + " (ID = " + id + ")").withBridge(bridge.getThing().getUID())
```

Comment:
As you set the sensor Id as a property as well, you should remove it from the label, but rather use `withRepresentationProperty("id")`, which indicated to the UI that this should be shown for identification.